packages/x-date-pickers/src/AdapterDateFns/AdapterDateFns.ts

```diff
@@ -0,0 +1,566 @@
+// @ts-ignore
+import longFormatters from 'date-fns/_lib/format/longFormatters';
+import { AdapterFormats, AdapterUnits, FieldFormatTokenMap, MuiPickersAdapter } from '../models';
+
+type DateFnsLocale = typeof defaultLocale;
+
+interface AdapterDateFnsOptions {
+  locale?: DateFnsLocale;
+  formats?: Partial<AdapterFormats>;
+}
+
+const formatTokenMap: FieldFormatTokenMap = {
+  // Year
+  y: { sectionType: 'year', contentType: 'digit', maxLength: 4 },
+  yy: 'year',
+  yyy: { sectionType: 'year', contentType: 'digit', maxLength: 4 },
+  yyyy: 'year',
+
+  // Month
+  M: { sectionType: 'month', contentType: 'digit', maxLength: 2 },
+  MM: 'month',
+  MMMM: { sectionType: 'month', contentType: 'letter' },
+  MMM: { sectionType: 'month', contentType: 'letter' },
+  L: { sectionType: 'month', contentType: 'digit', maxLength: 2 },
+  LL: 'month',
+  LLL: { sectionType: 'month', contentType: 'letter' },
+  LLLL: { sectionType: 'month', contentType: 'letter' },
+
+  // Day of the month
+  d: { sectionType: 'day', contentType: 'digit', maxLength: 2 },
+  dd: 'day',
+  do: { sectionType: 'day', contentType: 'digit-with-letter' },
+
+  // Day of the week
+  E: { sectionType: 'weekDay', contentType: 'letter' },
+  EE: { sectionType: 'weekDay', contentType: 'letter' },
+  EEE: { sectionType: 'weekDay', contentType: 'letter' },
+  EEEE: { sectionType: 'weekDay', contentType: 'letter' },
+  EEEEE: { sectionType: 'weekDay', contentType: 'letter' },
+  i: { sectionType: 'weekDay', contentType: 'digit', maxLength: 1 },
+  ii: 'weekDay',
+  iii: { sectionType: 'weekDay', contentType: 'letter' },
+  iiii: { sectionType: 'weekDay', contentType: 'letter' },
+  e: { sectionType: 'weekDay', contentType: 'digit', maxLength: 1 },
+  ee: 'weekDay',
+  eee: { sectionType: 'weekDay', contentType: 'letter' },
+  eeee: { sectionType: 'weekDay', contentType: 'letter' },
+  eeeee: { sectionType: 'weekDay', contentType: 'letter' },
+  eeeeee: { sectionType: 'weekDay', contentType: 'letter' },
+  c: { sectionType: 'weekDay', contentType: 'digit', maxLength: 1 },
+  cc: 'weekDay',
+  ccc: { sectionType: 'weekDay', contentType: 'letter' },
+  cccc: { sectionType: 'weekDay', contentType: 'letter' },
+  ccccc: { sectionType: 'weekDay', contentType: 'letter' },
+  cccccc: { sectionType: 'weekDay', contentType: 'letter' },
+
+  // Meridiem
+  a: 'meridiem',
+  aa: 'meridiem',
+  aaa: 'meridiem',
+
+  // Hours
+  H: { sectionType: 'hours', contentType: 'digit', maxLength: 2 },
+  HH: 'hours',
+  h: { sectionType: 'hours', contentType: 'digit', maxLength: 2 },
+  hh: 'hours',
+
+  // Minutes
+  m: { sectionType: 'minutes', contentType: 'digit', maxLength: 2 },
+  mm: 'minutes',
+
+  // Seconds
+  s: { sectionType: 'seconds', contentType: 'digit', maxLength: 2 },
+  ss: 'seconds',
+};
+
+const defaultFormats: AdapterFormats = {
+  dayOfMonth: 'd',
+  fullDate: 'PP',
+  fullDateWithWeekday: 'PPPP',
+  fullDateTime: 'PP p',
+  fullDateTime12h: 'PP hh:mm aa',
```

Comment:
The current adapter is using `aaa` instead of `aa` (`am/pm` instead of `AM/PM`) and I don't understand why...
Do you validate this change ? It makes the behavior a lot more coherent with the other adapters.

Reply:
I do agree that it makes little to no sense to have this meridiem format only for this adapter 🙈 👍 
Both Luxon and Moment are using the upper-case format.